software/control/core.py

```diff
@@ -775,6 +776,10 @@ def get_mm_per_ustep_Z(self):
     def set_flag_click_to_move(self, flag):
         self.click_to_move = flag
 
+    def wait_till_operation_is_completed(self):
+        while self.microcontroller.is_busy():
```

Comment:
I think there's already a `self.microcontroller.wait_till_operation_is_complete`.  Should we just use that here?

AKA:
```
def wait_till_operation_is_complete(self):
    return self.microcontroller.wait_till_operation_is_complete()
```

Reply:
@veerwang

Reply:
Sure, I would modify it.